03-rnaseq/ortholog-mapping_rnaseq_01_ensembl.Rmd

```diff
@@ -0,0 +1,369 @@
+This chunk of code will read the both TSV files and add them as data frames to your environment.
+
+```{r}
+# Read in metadata TSV file
+metadata <- readr::read_tsv(file.path(
+  data_dir, # Replace with path to your metadata file
+  "metadata_SRP070849.tsv" # Replace with the name of your metadata file
+))
+
+# Read in data TSV file
+df <- readr::read_tsv(file.path(
+  data_dir, # Replace with path to your data file
+  "SRP070849.tsv" # Replace with the name of your data file
+)) %>%
+  # Tuck away the gene ID column as rownames
+  tibble::column_to_rownames("Gene")
+```
+
+Let's ensure that the metadata and data are in the same sample order. 
+
+```{r}
+# Make the data in the order of the metadata
+df <- df %>%
+  dplyr::select(metadata$refinebio_accession_code)
+
+# Check if this is in the same order
+all.equal(colnames(df), metadata$refinebio_accession_code)
+
+# Bring back the "Gene" column in preparation for mapping
+df <- df %>%
+  tibble::rownames_to_column("Gene")
+```
+
+## Mapping mouse Ensembl gene IDs to human symbols
+
+refine.bio data uses Ensembl gene identifiers, which will be in the first column.
+
+```{r}
+# Let's take a look at the first 6 items in the `Gene` column of `df`
+head(df$Gene)
+```
+
+Ensembl gene identifiers have different species-specific prefixes. 
+In mouse, Ensembl gene identifiers begin with `ENSMUSG` (in human, `ENSG`, etc.).
+
+Now let's do the conversion!
+We'll use the `mouse` data from `hcop`; this is what `mouse` looks like.
+
+```{r}
+mouse
+```
+
+We're interested in the `human_symbol` and `mouse_ensembl` columns specifically.
+
+```{r}
+human_mouse_key <- mouse %>%
+  # we'll want to subset mouse to only the columns we're interested in
+  dplyr::select(mouse_ensembl, human_symbol)
+# Now let's join the expression data from `df` for the subset of genes in `df`
+human_mouse_mapped_df <- df %>%
+  dplyr::left_join(human_mouse_key, by = c("Gene" = "mouse_ensembl"))
+```
+
+Here's what the new data frame looks like:
+
+```{r}
+head(human_mouse_mapped_df, n = 25)
+```
+
+Let's get a summary of the gene symbols returned in the `human_symbol` column of our mapped data frame.
+
+```{r}
+# We can use the `count()` function to get a count of the distribution of values in the `human_symbol` column
+human_mouse_mapped_df %>%
+  dplyr::group_by(human_symbol) %>%
+  dplyr::count() %>%
+  # Sort by highest `n` which would be the human gene symbol with the most mapped mouse Ensembl IDs
+  dplyr::arrange(desc(n))
+```
+
+Looks like we have mapped symbols!
+We also have 18908 NAs, which is okay, because we do not expect everything to map across species.
+
+## Take a look at some multi-mappings
+
+If a mouse Ensembl gene ID maps to multiple human symbols, the gene expression values get duplicated.
+Let's look at the `ENSMUSG00000000037` example below.
+
+```{r}
+human_mouse_mapped_df %>%
+  dplyr::filter(Gene == "ENSMUSG00000000037") %>%
+  dplyr::select(Gene, human_symbol, dplyr::everything())
```

Comment:
I would say the same comment here as I did in the microarray PR, we should reorder this once in an earlier step so we don't have to do this twice.